runtime/src/main/java/org/corfudb/runtime/view/stream/AbstractQueuedStreamView.java

```diff
@@ -202,7 +202,16 @@ protected ILogData getNextEntry(QueuedStreamContext context,
      *
      * @param address       The address to read.
      */
-    protected abstract @Nonnull ILogData read(final long address);
+    protected @Nonnull ILogData read(final long address) {
+        try {
+            return readImpl(address);
```

Comment:
Initially I intended to catch and populate the trimmed exception in BackpointerStreamView, but then it didnt seem to make sense because BackpointerStreamView is the "concrete" implementation and seemed like it need to be in a higher layer. I don't like the xxImpl methods too, but that's just how the code is structured.

I can move it to BackpointerStreamView, I don't have a strong preference on this issue @medhavidhawan @dahliamalkhi what do you guys think?